katportalclient/client.py

```diff
@@ -1612,6 +1612,53 @@ def modify_userlog(self, userlog, tag_ids=None):
             method='POST', body=json.dumps(userlog))
         raise tornado.gen.Return(json.loads(response.body))
 
+    @tornado.gen.coroutine
+    def sensor_subarray_lookup(self, sub_nr, component, sensor,
+                               katcp_name=False):
+        """Return the full sensor name based on a generic component and sensor
+        name, for the given subarray.
+
+        This method gets the full sensor name based on a generic component and
+        sensor name, for a given subarray. This method will return a failed
+        katcp response if the given subarray is not in the 'active' or
+        'initialising' state.
+
+
+        .. note::
+
+            The websocket is not used for this request - it does not need
+            to be connected.
+
+        Parameters
+        ----------
+        sub_nr: int
+            The sub_nr on which to do the sensor lookup. The given component
+            must be assigned to this subarray for a successful lookup.
```

Comment:
Agree with @ajoubertza - In short order parameter descriptions according to their place in the def.
sub_nr should be optional and if not provided it should magically come from the sitemap. 
OR blow up if it was a sitemap without a sub_nr and non provided.